### DISCUSSION TITLE
Temporary projects for testing

### DISCUSSION THREAD

**Discussion by theetrain** [expo/expo#13861] [Proposals and Ideas]
We use Expo to test and build native apps, and we also use NextJS to build for web from the same codebase.

When a developer makes a pull request on our repository, that PR spins up a disposable testing environment for web by deploying the code from that PR to a temporary environment. When the PR merges, that temporary environment self-deletes.

It would be cool to achieve that same functionality with Expo: being able to test only the code changes made in a pull request in isolation. A couple of ideas I have are:

1. When a PR is created, expo-cli creates a new project using a slug based on the PR's git branch, and then publishes a build that can be tested via Expo GO using `https://expo.dev/@org/project-slug` URLs. When the PR is merged, the project can be removed using a cleanup task. **This would require expo-cli to support project deletion so that cleanup tasks can be made.**
2. A more refined, but admittedly more complex solution, is to support disposable build endpoints on a per-project basis within https://expo.dev/

This mainly benefits testers, making it easy for developers to make pull requests and share a URL for testers to use with Expo GO.

**Comment by barthap** [comment]
Maybe this would help? https://github.com/expo/expo-preview-action There's an example on how to set-up a PR preview.
